indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -103,7 +103,9 @@ public final static InputRow parseInputRow(Object value, InputRowParser parser)
   {
     if(parser instanceof StringInputRowParser && value instanceof Text) {
       //Note: This is to ensure backward compatibility with 0.7.0 and before
-      return ((StringInputRowParser)parser).parse(value.toString());
+      return ((StringInputRowParser) parser).parse(value.toString());
+    } else if(value instanceof InputRow) {
+      return (InputRow)value;
```

Comment:
hmmm, that sounds true. it gets a lil weird when "name" and "fieldName" in the aggregator are not same. I will need to think how to treat rows read from segment differently.
also it seems DatasourcePathSpec should really get list of metrics from "name" and not "fieldName" .